### PR TITLE
feat(buildkite): support git-hosted plugins

### PR DESCRIPTION
**Pull request by jhpierce**
## Changes:

Support updates for buildkite plugins hosted on Github.

## Context:

See [discussion](https://github.com/renovatebot/renovate/discussions/12984)

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by JamieMagee** [comment]
@jhpierce It looks like you need to set your git email correctly before you can sign the CLA?

**Comment by jhpierce** [comment]
Thank you! I think I did it?

File: lib/manager/buildkite/__fixtures__/pipeline3.yml [hunk 1]
```diff
@@ -8,14 +8,6 @@ steps:
...
   - wait
 
-  # Use the app image built above to run concurrent tests
-  - name: "Docker Test %n"
-    command: test.sh
-    parallelism: 25
-    plugins:
-      https://github.com/buildkite/plugin-docker-compose#v1.3.2:
-        run: app
-
```
Comment (rarkins) [inline lib/manager/buildkite/__fixtures__/pipeline3.yml:18]: Why was this removed?

Comment (jhpierce) [reply lib/manager/buildkite/__fixtures__/pipeline3.yml:18]: The test that uses that sample pipeline is testing whether the correct "skip reasons" are added. With the functionality I introduced, this plugin should no longer be skipped, so I figured it would make sense to remove it from the test.

**Review by rarkins — APPROVED** [review]
